cmd/backup/encrypt_archive.go

```diff
@@ -4,20 +4,87 @@
 package main
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path"
 
+	"github.com/ProtonMail/go-crypto/openpgp/armor"
 	openpgp "github.com/ProtonMail/go-crypto/openpgp/v2"
 	"github.com/offen/docker-volume-backup/internal/errwrap"
 )
 
-// encryptArchive encrypts the backup file using PGP and the configured passphrase.
-// In case no passphrase is given it returns early, leaving the backup file
+func readArmoredKeys(data []byte) (openpgp.EntityList, error) {
+	block, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+	return block, nil
+}
+
+func (s *script) encryptAsymmetrically(outFile *os.File) (io.WriteCloser, func() error, error) {
+	if s.c.GpgPublicKeys == "" {
+		return nil, nil, nil
+	}
+
+	entityList, err := readArmoredKeys([]byte(s.c.GpgPublicKeys))
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, fmt.Sprintf("Error parsing key: %v", err))
+	}
+
+	armoredWriter, err := armor.Encode(outFile, "PGP MESSAGE", nil)
+	if err != nil {
+		return nil, nil, errwrap.Wrap(err, "error preparing encryption")
+	}
+
+	_, name := path.Split(s.file)
+	dst, err := openpgp.Encrypt(armoredWriter, entityList, nil, nil, &openpgp.FileHints{
+		FileName: name,
+	}, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return dst, func() error {
+		dst.Close()
+		armoredWriter.Close()
+		return nil
+	}, err
+}
+
+func (s *script) encryptSymmetrically(outFile *os.File) (io.WriteCloser, func() error, error) {
+	if s.c.GpgPassphrase == "" {
+		return nil, nil, nil
+	}
+
+	_, name := path.Split(s.file)
+	dst, err := openpgp.SymmetricallyEncrypt(outFile, []byte(s.c.GpgPassphrase), &openpgp.FileHints{
+		FileName: name,
+	}, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return dst, dst.Close, nil
+}
+
+// encryptArchive encrypts the backup file using PGP and the configured passphrase or publickey(s).
+// In case no passphrase or publickey is given it returns early, leaving the backup file
 // untouched.
 func (s *script) encryptArchive() error {
-	if s.c.GpgPassphrase == "" {
+
+	var encrypt func(outFile *os.File) (io.WriteCloser, func() error, error)
+
+	switch {
+	case s.c.GpgPassphrase != "" && s.c.GpgPublicKeys != "":
+		return errors.New("error in selecting asymmetric and symmetric encryption methods: conflicting env vars are set")
```

Comment:
```suggestion
		return errwrap.Wrap(nil, "error in selecting asymmetric and symmetric encryption methods: conflicting env vars are set")
```
This is a bit non-obvious,but like this the error will start recording a poor man's stack trace for us.

Reply:
works for me